src/battle/battle_script_commands.c

```diff
@@ -1186,6 +1186,76 @@ BOOL btl_scr_cmd_54_ohko_move_handle(void *bw, struct BattleStruct *sp)
     return FALSE;
 }
 
+BOOL btl_scr_cmd_7c_beat_up_damage_calc(void *bw, struct BattleStruct *sp)
+{
+    int species, form;
+    s32 newBaseDamage;
+    struct BattlePokemon *mon;
+    
+    IncrementBattleScriptPtr(sp, 1);
+
+    int partyCount = GetPartySize(bw, sp->attack_client);
+
+    if (sp->multi_hit_count_temp == 0) {
+
+        sp->multi_hit_count_temp = 2;
+        sp->loop_hit_check = 0xFD;
+        sp->beat_up_count = 0;
+        mon = GetPartyMon(bw, sp->attack_client, sp->beat_up_count); 
+
+        while(!(sp->beat_up_count != sp->sel_mons_no[sp->attack_client] ||
+                (GetMonData(mon, ID_PARA_hp, 0) && 
+                GetMonData(mon, ID_PARA_monsno_egg, 0) && 
+                GetMonData(mon, ID_PARA_monsno_egg, 0) != 494 && 
+                GetMonData(mon, ID_PARA_condition, 0) == 0)))
+                {
+
+            sp->beat_up_count++;
+            mon = GetPartyMon(bw, sp->attack_client, sp->beat_up_count); 
+        
+        }
+    }   
+
+    mon = GetPartyMon(bw, sp->attack_client, sp->beat_up_count);
+    species = GetMonData(mon, ID_PARA_monsno, 0);
+    form = GetMonData(mon, ID_PARA_form_no, 0);
+
+    newBaseDamage = PokeFormNoPersonalParaGet(species, form, PERSONAL_BASE_ATTACK);
+    newBaseDamage /= 10;
+    sp->damage_power = sp->aiWorkTable.old_moveTbl[sp->current_move_index].power;
```

Comment:
this should be `sp->moveTbl[sp->current_move_index].power` due to expanded move table being moved out of the aiWorkTable structure

Reply:
Done